processing/src/main/java/org/apache/carbondata/processing/newflow/complexobjects/StructObject.java

```diff
@@ -0,0 +1,19 @@
+package org.apache.carbondata.processing.newflow.complexobjects;
+
+public class StructObject {
+
+  private Object[] data;
+
+  public StructObject(Object[] data) {
+    this.data = data;
+  }
+
+  public Object[] getData() {
+    return data;
+  }
+
+  public void setData(Object[] data) {
```

Comment:
I guess `setData` makes more sense as complete data of struct passed once instead one by one.